packages/astro/test/fixtures/astro-scripts/src/components/Inline.astro

```diff
@@ -1,3 +1,8 @@
 <script>
-  console.log('some content here.');
+	interface Props {
+		name: string;
+	}
```

Comment:
Ooh is the compiler ready for this? We follow [4.12.1.3](https://html.spec.whatwg.org/multipage/scripting.html#restrictions-for-contents-of-script-elements) so I think generics should be fine... 

Would be nice to test! LMK if you think that should be in the compiler rather than here.

Reply:
Not sure I follow, this is tested, this file is a test file.